sound/soc/sof/intel/hda-stream.c

```diff
@@ -105,11 +105,9 @@ int hda_dsp_stream_setup_bdl(struct snd_sof_dev *sdev,
 	stream->frags = 0;
 
 	/*
-	 * set IOC if don't use position IPC
-	 * and period_wakeup needed.
+	 * set IOC if disalbe nowakeup mode and period_wakeup is needed
 	 */
-	ioc = hda->no_ipc_position ?
-	      !stream->no_period_wakeup : 0;
+	ioc = !stream->no_period_wakeup;
```

Comment:
I was confused this line too. IOC is set when periodic wakeup is needed. but it was a bit more confused.

- SND_SOC_SOF_DEBUG_FORCE_IPC_POSITION is eanbled, no_ipc_position is forcely set 0. then IOC is always 0
- SND_SOC_SOF_DEBUG_FORCE_IPC_POSITION is disabled, NO_WAKEUP case(no_period_wakeup=1), IOC will be 0,
- SND_SOC_SOF_DEBUG_FORCE_IPC_POSITION is disabled, WAKEUP case(no_period_wakeup=0), IOC will be 1,

I could put featurization, and make it same too. but last proposal is simple and clear to me. any concern or suggestion?
#if SND_SOC_SOF_DEBUG_DISABLE_NOWAKEUP
#else
#endif

Reply:
ah... just got remembered. SKL- case, it should be WAKEUP mode. It was depending on pcm_trigger pointer. when it is null, we have to set IOC =1. I will wait for Pierre's review. I will consolidate all review comments.